Sources/Prometheus/Counter.swift

```diff
@@ -0,0 +1,90 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftPrometheus open source project
+//
+// Copyright (c) 2018-2023 SwiftPrometheus project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftPrometheus project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Atomics
+import CoreMetrics
+
+public final class Counter: Sendable {
```

Comment:
Oh we require that all metric backing handlers are classes? This is a bit sad since this could just be a struct with reference semantics.

Second question, does this need to be public?

Reply:
> Oh we require that all metric backing handlers are classes? This is a bit sad since this could just be a struct with reference semantics.

I eventually want to change swift-metrics to not require AnyObject anymore. This will make NoOps much cheaper.

> Second question, does this need to be public?

Yes. I think users should be able to use the Prometheus lib without going through swift-metrics.

Reply:
Agreed on both, eventually we can do a revamp of metrics lib. It's one of the earliest and we've learned much since.

Yes, types should be public, prom has more detailed semantics than swift metrics so an "app" can definitely use it direcly